pkg/service/rtcservice.go

```diff
@@ -72,11 +72,21 @@ func (s *RTCService) validate(r *http.Request) (string, routing.ParticipantInit,
 	roomName := r.FormValue("room")
 	reconnectParam := r.FormValue("reconnect")
 	autoSubParam := r.FormValue("auto_subscribe")
+	publishParam := r.FormValue("publish")
 
 	if onlyName != "" {
 		roomName = onlyName
 	}
 
+	// this is new connection for existing participant -  with publish only permissions
+	if publishParam != "" {
+		// Make sure grant does not have 'subscribe' set,
+		if claims.Video.CanSubscribe != nil && *claims.Video.CanSubscribe == true {
```

Comment:
can we set the grant instead of failing the request? since the same token is used, it will likely be set to true.